Mathlib/LinearAlgebra/Finsupp/LinearCombination.lean

```diff
@@ -258,6 +249,41 @@ theorem linearCombination_onFinset {s : Finset α} {f : α → R} (g : α → M)
   contrapose! h
   simp [h]
 
+variable [Module S M] [SMulCommClass R S M]
+
+variable (S) in
+/-- `Finsupp.bilinearCombination R S v f` is the linear combination of vectors in `v` with weights
+in `f`.
+
+This map is linear in `v` if `R` is commutative, and always linear in `f`.
+See note [bundled maps over different rings] for why separate `R` and `S` semirings are used.
+-/
+def bilinearCombination : (α → M) →ₗ[S] (α →₀ R) →ₗ[R] M where
+  toFun v := linearCombination R v
+  map_add' u v := by ext; simp [Finset.sum_add_distrib, Pi.add_apply, smul_add]
+  map_smul' r v := by ext; simp [Finset.smul_sum, smul_comm]
+
+theorem bilinearCombination_apply_eq_linearCombination :
+    bilinearCombination R S v = linearCombination R v :=
+  rfl
+
+theorem bilinearCombination_apply (f) :
+    bilinearCombination R S v f = f.sum fun i c ↦ c • v i := by
+  rfl
+
+variable (α M)
+
+@[simp]
+theorem linearCombination_zero : linearCombination R (0 : α → M) = 0 :=
```

Comment:
I don't understand why you're moving these

Reply:
It was a mistake.

Reply:
I moved `linearCombination_zero` to its initial place, but the `bilinearCombination` part fits better at the end, because it uses some `simp` lemmas that are proved earlier and requires to add some variables (such as `Module S M`).